### DISCUSSION TITLE
Flask should be 1º class fully functional subsystem of connexion 3.x.x

### DISCUSSION THREAD

**Discussion by carloscbl** [spec-first/connexion#1959] [Polls]
Difference i can see from connexion 2.0 to 3.0 flask is no longer a fully fledged subsistem but a subset of elements.
This means flask/werkzeug is not functional by themself after you start "FlaskApplication".

EG:

1. We are not able to use test_client from flask, we instead need to use connexion test_client -> problem with this is they don't share same API and would be too much while keeping compat with starlete subsystem?
2. Also flask/werkzeug by itself will be missing current_app context which breaks many features and libs for flask "flask limiter" as eg

To my opinion:
Connexion is a beloved lib but it lacks resources like Flask/werkzeug does, so would be for connexion too burdensome to not fully take advance and be fully compatible with flask, we can have small quirks but we should support the main flask features.

**Comment by RobbeSneyders** [comment]
Connexion 3 was redesigned due to multiple reasons explained in [our blogpost](https://medium.com/@robbe.sneyders/redesigning-connexion-as-an-asgi-middleware-stack-a5dc17e81ff8).

Just like Flask has an ecosystem of extensions, ASGI has an ecosystem of middleware. Eg [asgi-ratelimit](https://github.com/abersheeran/asgi-ratelimit) as replacement for `Flask-Limiter`.
